probe/endpoint/procspy/background_reader_linux.go

```diff
@@ -38,6 +38,28 @@ func newBackgroundReader(walker process.Walker) *backgroundReader {
 	return br
 }
 
+func newForegroundReader(walker process.Walker) *backgroundReader {
```

Comment:
This shouldn't be here (either change the file name or place it somewhere else).